### ISSUE TITLE
Filter data on MS2 mz range

### ISSUE THREAD

**Issue by jorainer** [EuracBiomedicalResearch/CompoundDb#29]
This is related to issue #28, specifically the point 1) described there. Retrieve data based on the m/z range of their MSMS spectra.
- Add columns `msms_mz_range_min` and `msms_mz_range_max` to the *msms_spectrum* database table.
- Add `MsmsMzRangeMinFilter` and `MsmsMzRangeMaxFilter` filters (terrible name but we have to be compliant with the naming scheme for AnnotationFilter classes).

Use case 1: get all MSMS spectra with all of their peaks within a m/z range:
```r
spectra(compdb, filter = ~ msms_mz_range_min > 123.4 & msms_mz_range_max < 343.4)
```

Use case 2: get all MSMS spectra with a peak potentially within an m/z range (that's what we need for 1) in issue #28):
```r
spectra(compdb, filter = ~ msms_mz_range_min <= 123.4 & msms_mz_range_max >= 123.5)
```

**Comment by jotsetung** [comment]
Now proceeding with finding the spectra that have an actual m/z matching `195.08` (using the `hasMz` method from #28).

```r
system.time(all_has_pk <- hasMz(all, mz = mz_in, ppm = 20))
   user  system elapsed 
 10.930   0.203  11.212 

system.time(subs_has_pk <- hasMz(subs, mz = mz_in, ppm = 20))
   user  system elapsed 
  4.058   0.044   4.128 

sum(all_has_pk)
[1] 591

sum(sus_has_pk)
[1] 590
```

The same spectra were detected, with the exception of one, that was missed because the filter with `msms_mz_range_min` and `msms_mz_range_max` did not consider a `ppm`.
